core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -171,6 +172,20 @@ public static Object[][] differentValues()
                 };
     }
 
+    @Test(dataProvider = "types")
+    public void testMultipleRleWithTheSameValueProduceRle(Type type)
+    {
+        PositionsAppender positionsAppender = POSITIONS_APPENDER_FACTORY.create(type, 10, DEFAULT_MAX_PAGE_SIZE_IN_BYTES);
+
+        Block value = notNullBlock(type, 1);
+        positionsAppender.append(allPositions(3), rleBlock(value, 3));
+        positionsAppender.append(allPositions(2), rleBlock(value, 2));
+
+        Block actual = positionsAppender.build();
+        assertEquals(actual.getPositionCount(), 5);
+        assertTrue(actual instanceof RunLengthEncodedBlock, actual.toString());
```

Comment:
Maybe you could use `io.airlift.testing.Assertions#assertInstanceOf(T, java.lang.Class<U>)`?